src/api/app/controllers/webui/users/notifications_controller.rb

```diff
@@ -27,6 +27,7 @@ def update
     # rubocop:enable Rails/SkipsModelValidations
 
     # manually update the count and the filtered subset after the update
+    set_unread_notifications
```

Comment:
We have to set `@unread_notifications` in the `update` case. Otherwise, `@counted_notifications['unread']` will be assigned later to the value previous to the `.update_all(...)` call.

@hennevogel please take a look again.

Reply:
It might be nice to tell people where this comes from in a comment...

Reply:
> It might be nice to tell people where this comes from in a comment. Also the instance variable...

True. I added a couple of comments.